integration/src/main/java/scripts/testscripts/WorkspaceLifecycle.java

```diff
@@ -40,11 +43,24 @@ public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspace
     String validUserFacingId = "user-facing-id-" + uuidStr;
     String validUserFacingId2 = "user-facing-id-2-" + uuidStr;
 
+    Property property1 = new Property();
```

Comment:
nit: can we do a bit refactoring to make this a bit more readable :) 

you can have a helper method `Properties buildProperties(Map<String, String> propertyMap)`. And then to initialize a Map you can do a one liner with Map.of().At line 118 we will do `buildProperties(propertyMap.putAll(updateProperties))` so it's clear that the update is following the same rule as java hashmap#putAll.

Reply:
Hi Yu. I find the Map.of is an unmodifiable map. So it will fail while I assert the updated properties if I use putAll to combine. 
I just modify the all first propertymap to modifiable map. Just a head up.